src/Gallery.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react'
+import { STATUS } from './constants';
+import { setTimeout } from 'timers';
+
+class Gallery extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            cats: [],
+            loadingState: STATUS.FETCHING,
+            autoScroll: false,
+        }
+
+        this.setScroll = this.setScroll.bind(this);
+        this.handleScroll = this.handleScroll.bind(this);
+        this.scrollToBottom = this.scrollToBottom.bind(this);
+        this.handlecheckbox = this.handlecheckbox.bind(this);
+        this.fetchCatsCollection = this.fetchCatsCollection.bind(this);
+    };
+
+    componentDidMount() {
+        this.fetchCatsCollection();
+        window.addEventListener('scroll', this.handleScroll)
+    }
+
+    componentWillUnmount(){
+        
+    }
+
+    scrollToBottom() {
+        this.pageEnd.scrollIntoView({ behavior: "smooth" });
+    }
+
+    handleScroll(e) {
```

Comment:
this should be debounced, as it will fire multiple times once user reaches the bottom of the page.